src/Domain/Amr/Helpers.hpp

```diff
@@ -43,6 +44,14 @@ std::array<size_t, VolumeDim> desired_refinement_levels_of_neighbor(
     const std::array<Flag, VolumeDim>& neighbor_flags,
     const OrientationMap<VolumeDim>& orientation);
 
+/// \ingroup AmrGroup
+/// Fraction of the logical volume of a block covered by an element
+///
+/// \note The sum of this over all the elements of a block should be one
+template <size_t VolumeDim>
+boost::rational<size_t> fraction_of_block_volume(
```

Comment:
std::ratio is a compile time rational

Reply:
Oh right 👍

Reply:
What about `Utilities/Rational.hpp`, which we have because the boost one is really slow?

Reply:
This is only used for a sanity check reduction, so speed shouldn't be an issue.

When I add the reduction action using this I'll try out using `Rational` instead of `boost::rational`